### PR TITLE
Readme overhaul and associated py3 fixes

### PR DESCRIPTION
**Pull request by gileri**
I tried following the README.md on Ubuntu 16.04 and 18.04 and hit some pitfalls.
Some webpack-built assets changed path, so gitignore reflects that.

### CODE REVIEW

**Comment by frodrigo** [comment]
Thank you for your review. We just change the code layout and not yet update the doc. Note the preferred way to install it now with docker.

File: README.md [hunk 1]
```diff
@@ -12,53 +12,72 @@ For running Osmose Frontend with Docker go into the docker directory. You can al
 Manual Installation
 -------------------
 
+### Dependencies
+
+For Ubuntu Server 16.04 and 18.04
+
+```
+apt install python2.7 python2.7-dev virtualenv gcc pkg-config libpng-dev libjpeg-dev libfreetype6-dev postgresql-server-dev-all libgeos-dev g++ python-shapely nodejs npm
```
Comment (frodrigo) [inline README.md:20]: This week we just switch to python3.

File: README.md [hunk 2]
```diff
@@ -12,53 +12,72 @@ For running Osmose Frontend with Docker go into the docker directory. You can al
...
+#### Database
+
+```
+apt install postgresql postgresql-contrib
+```
+
+#### Web server
+
+```
+apt install apache2 libapache2-mod-wsgi
```
Comment (frodrigo) [inline README.md:32]: libapache2-mod-wsgi-py3 now

File: README.md [hunk 3]
```diff
@@ -12,53 +12,72 @@ For running Osmose Frontend with Docker go into the docker directory. You can al
...
-
 Create a python virtualenv, active it and install python dependencies
 ```
 virtualenv --python=python2.7 osmose-frontend-venv
 source osmose-frontend-venv/bin/activate
-pip install -r requirements.txt
 ```
 
-Generate markers
+Workaround for Ubuntu 16.04, as pip install the [latest setuptools](https://github.com/tilezen/mapbox-vector-tile/blob/e16fca20ec4e45e12caff8393150ff3327ad1a4d/setup.py#L33),
```
Comment (frodrigo) [inline README.md:59]: We are in 2020, I think we can drop 16.04 support.

File: README.md [hunk 4]
```diff
@@ -68,22 +87,24 @@ createdb -E UTF8 -T template0 -O osmose osmose_frontend
 psql -c "CREATE EXTENSION pgcrypto" osmose_frontend
 ```
 
-As normal user, create the database tables:
+Go back to your original user and create the database tables. Replace `127.0.0.1` by your database hostname or IP if necessary
```
Comment (frodrigo) [inline README.md:90]: It should work using unix socket, no need of IP.

**Comment by gileri** [comment]
> Note the preferred way to install it now with docker.

Yeah that's what I'm using. But these instructions seem necessary to me for anyone not using Docker/containers, or even Docker packagers. Do you think they should be removed in favor of only Dockerfiles ?

File: README.md (hunk 4 above)
Comment (gileri) [reply README.md:90]: That is a workaround as postgresql may be configured to use `peer` authentication for socket-based connections. Using TCP forces using login/password authentication, and will match "production" environment setup, that is docker-based or a remote database server.

If you feel that explaining that here is better I can do that.

File: README.md (hunk 3 above)
Comment (gileri) [reply README.md:59]: Ok I will drop 16.04 support from the README.

File: README.md (hunk 1 above)
Comment (gileri) [reply README.md:20]: I've seen your commits, thank you for this switch ! I didn't realize that you were close to completion, congrats !

Should I update this PR to only use python 3 instructions then ?

Comment (frodrigo) [reply README.md:20]: If you wish to.

**Comment by frodrigo** [comment]
For the backend we move the manual install to a secondary file and no more in the main README.

File: README.md [hunk 5]
```diff
@@ -6,116 +6,4 @@ on a map.
...
-Database translations
----------------------
-
-When some issues are in the database, to get translations
-```
-cd tools/database/ && ./categ_menu_update.sh && ./item_menu_update.sh
-```
-
-Add "tools/cron.sh" to crontab, to run once per day.
+To run Osmose Frontend with Docker follow the [documentation in the docker directory](docker/README.md). You can also install the Frontend by following the [install guide](INSTALL.md)
```
Comment (frodrigo) [inline README.md:9]:
```suggestion
To run Osmose Frontend preferably with Docker follow the [documentation in the docker directory](docker/README.md). You can also install the Frontend manually by following the [install guide](INSTALL.md)
```

File: INSTALL.md [hunk 6]
```diff
@@ -1,9 +1,9 @@
 Manual Installation
 -------------------
 
-### Dependencies
+This guide is compatible with Ubuntu 18.04 and Ubuntu 20.04
```
Comment (jocelynj) [inline INSTALL.md:4]: In fact, we are using Debian on our infrastructure, but we can keep Ubuntu.

By the way, how do you know if it is compatible with Ubuntu 20.04, as it wasn't yet released?

File: README.md [hunk 7]
```diff
@@ -90,6 +85,10 @@ psql osmose_frontend -f tools/database/schema.sql
 
 Check data base parameter into `tools/utils.py`.
 
+### Generate markers
+```
+cd tools && ./make-markers.py
```
Comment (jocelynj) [inline README.md]: We need in fact to complete the command, as it is not sufficient to update markers.

I also run:
```(cd web && npm run build) && touch osmose.wsgi

File: README.md [hunk 8]
```diff
@@ -17,7 +17,7 @@ Manual Installation
 For Ubuntu Server 16.04
 
 ```
-apt install python2.7 python2.7-dev virtualenv gcc pkg-config libpng-dev libjpeg-dev libfreetype6-dev postgresql-server-dev-all libgeos-dev g++ python-shapely nodejs
+apt install python2.7 python2.7-dev virtualenv gcc pkg-config libpng-dev libjpeg-dev libfreetype6-dev postgresql-server-dev-all libgeos-dev g++ python-shapely nodejs npm
```
Comment (jocelynj) [inline README.md]: We don't have this package on our infrastructure - is it necessary on recent distrib versions?

File: INSTALL.md (hunk 6 above)
Comment (gileri) [reply INSTALL.md:4]: Ubuntu 20.04 beta was released, I followed the procedure on an [ubuntu:20.04](https://github.com/tianon/docker-brew-ubuntu-core/blob/857debd4347496c541f9806c084710dace49cd27/focal/Dockerfile) docker image.

I can also try on recent debian versions if necessary.

File: README.md (hunk 7 above)
Comment (gileri) [reply README.md]: I'm not sure I understand. `npm run build` is covered in the Javascript section. I can merge both sections if that's what you meant ?

File: README.md (hunk 8 above)
Comment (gileri) [reply README.md]: I didn't check on Debian, but on Ubuntu it is necessary.

Based [on that](https://github.com/osm-fr/osmose-frontend/pull/237#pullrequestreview-392116772) I will update this section.

File: README.md (hunk 7 above)
Comment (frodrigo) [reply README.md]: make-maker create new images, and need to be bundled by  `npm run build`

Comment (gileri) [reply README.md]: [Right](https://github.com/osm-fr/osmose-frontend/blob/master/web/webpack.config.js#L47-L48), nice catch !

I moved asset generation after `make-markers`, but didn't merge both sections as `webpack` also build JS and CSS, that are unrelated to markers I believe.

File: README.md (hunk 4 above)
Comment (gileri) [reply README.md]: Without answer I remove postgresql setup specific params/environment variables.

**Comment by frodrigo** [comment]
Thank you!

**Comment by gileri** [comment]
Thank you for merging :)

**Pull request merged**